examples/admin_create_topic/admin_create_topic.go

```diff
@@ -0,0 +1,93 @@
+// Create topic
+package main
+
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import (
+	"context"
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+
+	if len(os.Args) != 5 {
+		fmt.Fprintf(os.Stderr,
+			"Usage: %s <broker> <topic> <partition-count> <replication-factor>\n",
+			os.Args[0])
+		os.Exit(1)
+	}
+
+	broker := os.Args[1]
+	topic := os.Args[2]
+	numParts, err := strconv.Atoi(os.Args[3])
+	if err != nil {
+		fmt.Printf("Invalid partition count: %s: %v\n", os.Args[3], err)
+		os.Exit(1)
+	}
+	replicationFactor, err := strconv.Atoi(os.Args[4])
+	if err != nil {
+		fmt.Printf("Invalid replication factor: %s: %v\n", os.Args[4], err)
+		os.Exit(1)
+	}
+
+	// Create a new AdminClient.
+	// AdminClient can also be instantiated using an existing
+	// Producer or Consumer instance, see NewAdminClientFromProducer and
+	// NewAdminClientFromConsumer.
+	a, err := kafka.NewAdminClient(&kafka.ConfigMap{"bootstrap.servers": broker})
+	if err != nil {
+		fmt.Printf("Failed to create Admin client: %s\n", err)
+		os.Exit(1)
+	}
+
+	// Contexts are used to abort or limit the amount of time
+	// the Admin call blocks waiting for a result.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Create topics on cluster.
+	// Set Admin options to wait for the operation to finish (or at most 60s)
+	maxDur, err := time.ParseDuration("60s")
+	if err != nil {
+		panic("ParseDuration(60s)")
+	}
+	results, err := a.CreateTopics(
+		ctx,
```

Comment:
Just wanted to check how sure you are that a blocking call with `ctx` as the first argument is the most appropriate API for this in golang. This is a n00b question, just calling out something I was wondering about: having a context parameter makes usage of the api more verbose than it otherwise needs to be for those who don't want it. On the other hand, it does look like it's a new feature and may be an impending standard sort of thing to do?. But standard file io stuff doesn't seem to do this, and there seem to be standard patterns for using go routines to asyncify synchronous methods. Is that not good enough?

Reply:
The introduction of contexts was after a discussion with @travisjeffery, it seems to be the way forward for the Go community and we should follow suite.

Contexts provides a standardised way to limit or cancel long running blocking operations.

Reply:
it's a pity that in golang this convention necessarily add verbosity - in other languages this is not the case. but it seems the right choice as far as i can tell.

Reply:
Yeah, Golang seems to be quite explicit and verbose in these regards.